Free draw action action setup with boomenu and instructional toast

BarAgam opened the pull request.

barmako committed:
Removed used classes.
Refactor packages hierarchy for map and alerts.

barmako committed:
Refactored navigator to hold activity.
Added boom menu button for dynamic layers.

barmako committed:
Starting dynamic layer editing action when clicking boom-menu's button.
Showing an instructional toast on action start.
Stopped action fragments from showing soft keyboard when it starts.

barmako committed:
Changing names from dynamic layer to free-drawing

teamagam5 commented on app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawActionFragment.java:
```diff
@@ -0,0 +1,79 @@
+  @Inject
+  FreedrawViewModelFactory mFreedrawViewModelFactory;
+
+  @BindView(R.id.dynamic_layer_edit_map)
```
I think this id is influenced by your mistake of what u r doing :rofl:

teamagam5 commented on app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreeDrawActionFragment.java:
```diff
@@ -0,0 +1,79 @@
+  private void displayTwoFingersToast() {
+    LayoutInflater layoutInflater = getActivity().getLayoutInflater();
+
+    View toastLayout = layoutInflater.inflate(R.layout.toast_two_fingers,
```
extract method getToastLayout?

teamagam5 commented on app/src/main/java/com/teamagam/gimelgimel/app/map/actions/freedraw/FreedrawViewModel.java:
```diff
@@ -0,0 +1,39 @@
+
+  }
+
+  public void onSendFabClicked() {
```
not needed (same for the description methods)

teamagam5 commented on app/src/main/res/layout/fragment_free_draw.xml:
```diff
@@ -0,0 +1,99 @@
+          android:layout_width="match_parent"
+          android:layout_height="match_parent"
+          />
+      <android.support.design.widget.FloatingActionButton
```
remove

teamagam5 commented on app/src/main/res/layout/fragment_free_draw.xml:
```diff
@@ -0,0 +1,99 @@
+          app:srcCompat="@drawable/ic_undo"
+          />
+
+      <EditText
```
remove

teamagam5 commented on app/src/main/res/values-iw/strings.xml:
```diff
@@ -78,4 +78,12 @@
+  <string name="pref_notifications_header_title">התראות</string>
+  <string name="toast_two_fingers_text">הזז את המפה באמצעות שתי אצבעות</string>
+  <string name="pref_notifications_alerts_only_title">הצג התראות דחופות בלבד</string>
 </resources>
```
Reorganize this file. You have some weird things going on here... Duplicates etc.

teamagam5 commented on app/src/main/res/values/strings.xml:
```diff
@@ -163,6 +164,8 @@
   <string name="message_notification_summary">Message</string>
   <string name="message_notification_text">New incoming message</string>
   <string name="pref_general_about_title">About</string>
+  <string name="free_draw_title">Free draw</string>
```
One can be a reference to another:
`menu_action_free_draw_title` & `free_draw_title`
(Unless it is intended for separation)

teamagam5 commented on app/src/main/res/layout/fragment_free_draw.xml:
```diff
@@ -0,0 +1,99 @@
+        >
+
+      <android.support.v7.widget.AppCompatImageView
+          android:id="@+id/send_geometry_undo_image"
```
rename id

barmako committed:
CR fixes-
1. Initializing map viewmodel to display entities
2. Removed unnecessary views from free draw view (and their view-model corresponding methods)
3. Added two stub view for free draw (and their view-model corresponding methods)
4. removed unused strings from hebrew string resource file
